src/types.ts

```diff
@@ -580,6 +580,17 @@ export type ExtractSchema<T> = UnionToIntersection<
   T extends Hono<infer _, infer S, any> ? S : never
 >
 
+export type RequiredKeysOf<BaseType extends object> = Exclude<
```

Comment:
Please place `RequiredKeysOf` and `HasRequiredKeys` in `src/utils/types.ts` as they may be used in other places.